src/stores/activitiesStores/Finalize.js

```diff
@@ -0,0 +1,77 @@
+import { observable, action, runInAction, reaction } from 'mobx';
+import _ from 'lodash';
+import { Token, OracleStatus, AppLocation, SortBy } from '../../constants';
+import { queryAllOracles } from '../../network/graphQuery';
+import Oracle from '../models/Oracle';
+
+const INIT = {
+  loaded: false, // loading state?
+  loadingMore: false, // for laoding icon?
+  list: [], // data list
+  hasMore: true, // has more data to fetch?
+  offset: 0, // skip
+  limit: 16, // loading batch amount
+};
+
+export default class {
+  @observable loaded = INIT.loaded
+  @observable loadingMore = INIT.loadingMore
+  @observable list = INIT.list
+  @observable hasMore = INIT.hasMore
+  @observable offset = INIT.offset
+  limit = INIT.limit
+
+  constructor(app) {
+    this.app = app;
+    reaction(
+      () => this.list,
+      () => {
+        if (this.loaded && this.list.length <= this.offset) this.hasMore = false;
```

Comment:
shouldnt it be `this.list.length < this.offset`? if the length of the list matches the offset, that means there could be more to fetch. if it is under then we know that is the last page.

Reply:
good catch. thanks